app/Http/Controllers/FrontendTransportController.php

```diff
@@ -137,9 +138,9 @@ public function TrainTrip(Request $request): Renderable|RedirectResponse {
 
             $encounteredStart = false;
             $stopovers = $trip->stopovers
-                ->filter(function(Stopover $stopover) use ($departure, $startStation, &$encounteredStart): bool {
+                ->filter(function(Stopover $stopover) use ($startStation, &$encounteredStart): bool {
                     if (!$encounteredStart) { // this assumes stopovers being ordered correctly
-                        $encounteredStart = $stopover->departure_planned == $departure && $stopover->station->is($startStation);
```

Comment:
Removing timestamps from station checks is not a good idea, while this will break ring lines (not checked, but in my experience this is the case every time...)

Reply:
I suppose you mean a scenario like this:

![Screenshot 2024-03-10 at 19 15 12](https://github.com/Traewelling/traewelling/assets/36338179/3879e676-3235-44b8-82e2-e93324cee82a)

That example works without the timestamp check based on the planned departure of the stopover. The filter only removes stopovers before the first occurrence of the station. After that it stops filtering out stations. Duplicate occurrences of the same station should, therefore, be fine. I might be thinking of the wrong scenario here though 😅

Reply:
Hmmm... I've checked it with the S41/S42 from Berlin Gesundbrunnen and it seems to work just fine. I guess we'll roll with it and revert it in a worst case scenario? :D

Reply:
Sure. :D

Reply:
@DC7IA do you have some input here before we are merging this as you worked on this snipped before?

Reply:
@d3d9 do you have some input here before we are merging this as you worked on this snipped before?

Reply:
Reverting if something breaks is fine with me (of course). I'll mark this as resolved but feel free to gather more input if you like 🙂

Reply:
> @d3d9 do you have some input here before we are merging this as you worked on this snipped before?

Thank you for notifying me. I'm afraid I wasn't able to respond earlier.
I think the departure time check was there for a good reason. On my first attempt testing out the new code, I could notice how, also using the Gesundbrunnen example, with a train that is passing it the second time, I get shown the stopovers before the second Gesundbrunnen one. Checking in fails ("Departure time must be before arrival time"), but we should prevent the wrong stopovers being displayed.  
Instead of removing the departure time check, it should be kept and fixed with respect to timezones.